src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -412,6 +413,55 @@ def with_metadata(
         ]
         return self
 
+    def format(
+        self, format: Literal["csv", "json", "avro", "parquet", "orc", "xml"]
+    ) -> "DataFrameReader":
+        """Specify the format of the file(s) to load.
+
+        Args:
+            format: The format of the file(s) to load. Supported formats are csv, json, avro, parquet, orc, and xml.
+
+        Returns:
+            a :class:`DataFrameReader` instance that is set up to load data from the specified file format in a Snowflake stage.
+        """
+        canon_format = format.strip().lower()
+        allowed_formats = ["csv", "json", "avro", "parquet", "orc", "xml"]
+        if canon_format not in allowed_formats:
+            raise ValueError(
+                f"Invalid format '{format}'. Supported formats are {allowed_formats}."
+            )
```

Comment:
It might be cleaner to push validation logic into a property setter in case anything ends up setting self._format in the future.

Reply:
I updated both file using a setter. let me know if the new changes look good.

Reply:
Looks good to me.